Checkout/Checkout/Source/Validation/Validators/CardNumberValidator.swift

```diff
@@ -89,12 +89,21 @@ final class CardNumberValidator: CardNumberValidating {
   ) -> Card.Scheme? {
     let range = NSRange(location: 0, length: cardNumber.utf16.count)
 
-    return Card.Scheme.allCases.first { scheme in
+    let matchedScheme = Card.Scheme.allCases.first { scheme in
       cardSchemeToRegex(scheme)?.firstMatch(in: cardNumber, options: [], range: range) != nil
     }
+    return addSchemePropertyIfNeeded(scheme: matchedScheme, cardNumber: cardNumber)
   }
 
   private func validateDigitsOnly(in string: String) -> Bool {
     return CharacterSet.decimalDigits.isSuperset(of: CharacterSet(charactersIn: string))
   }
+    
+  private func addSchemePropertyIfNeeded(scheme: Card.Scheme?, cardNumber: String) -> Card.Scheme? {
+    if scheme == .maestro(length: 0) {
+        return .maestro(length: cardNumber.filter { Int("\($0)") != nil }.count)
```

Comment:
we've already checked for whitespace and non-digit characters. We can remove this extraneous check
```suggestion
        return .maestro(length: cardNumber.count)
```

Reply:
They are not actually being validated to the same standard, just removing white spaces. I have moved this validation to the callers of this method, so as suggested I did remove this filter here (you were correct), but had to extend that filter.

Working on issuing, I can see how a consumer may want to display input numbers in many formats:
```
4242 4242 4242 4242
4242 - 4242 - 4242 - 4242
4242 : 4242 : 4242 : 4242
4242 . 4242 . 4242 . 4242
```

Only the first one was being stripped correctly, others would result in false results.

Reply:
@harry-brown-cko I've had to revert this change, the behaviour was not quite the same. The original validation is looking for invalid characters and will issue a rejection with a reason _invalidCharacters_ if such an occurange is encountered.

This function on the other hand attaches the counter to number of Int's found inside the provided String to correctly sync maestro length. Whilst current flows do have them behave similarly, the expectation of this function is different and merging them is not really appropriate.